tests/PPC/TestLifting.cpp

```diff
@@ -0,0 +1,800 @@
+/*
+ * Copyright (c) 2022 Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <llvm/IR/LLVMContext.h>
+#include <remill/Arch/Arch.h>
+#include <remill/Arch/Instruction.h>
+#include <remill/Arch/Name.h>
+#include <remill/Arch/PPC/Runtime/State.h>
+#include <remill/BC/Optimizer.h>
+#include <remill/BC/Version.h>
+#include <remill/OS/OS.h>
+#include <test_runner/TestOutputSpec.h>
+#include <test_runner/TestRunner.h>
+
+#include <unordered_map>
+
+namespace {
+
+const static std::unordered_map<std::string,
+                                std::function<std::any(PPCState &)>>
```

Comment:
In `RegisterPrecondition`, we use `std::variant<uint64_t, uint8_t>` to represent a register type. Can we:

- Put `std::variant<uint64_t, uint8_t>` behind an alias?
- Use that type here instead of `std::any`?